app/javascript/components/Survey.js

```diff
@@ -31,17 +35,30 @@ const Survey = () => {
         }
     }
 
+    const handleOpen = () => {
+        setOpen(true);
+    }
+
+    const handleClose = () => {
+        setOpen(false);
+        navigate("/")
+    }
+
     useEffect(async () => {
         setBaseUrl(window.location.origin.replace(/\/#.*/, ""));
-        await createResponder();
         await fetch(`${baseUrl}/api/v1/surveys/${surveyId}`, {
             method: 'GET',
             headers: {
                 'Content-Type': 'application/json'
             },
         })
         .then(checkRequest)
-        .then(data => {
+        .then(async (data) => {
+            console.log(data.survey.isLive)
+            if(!data.survey.isLive) {
+                handleOpen();
+            }
+            await createResponder();
```

Comment:
to be safe, the next three lines should be in an else statement, also remove the console.log on line 57